manifests/enterprise.pp

```diff
@@ -208,7 +208,7 @@
   Optional[String[1]] $package_provider      = $splunk::params::package_provider,
   Boolean $manage_package_source             = true,
   Optional[String[1]] $package_source        = undef,
-  Array[String[1]] $install_options          = $splunk::params::enterprise_install_options,
+  Variant[Tuple[Hash,Hash,String,3,default], Array[String[1]]] $install_options = $splunk::params::enterprise_install_options,
```

Comment:
can you please align the `=`? The structure in the Hashes and the string length could also be enforced, should this be a custom datatype?

Reply:
I'll align the `=` in the next commit.

The Tuple layout is dependent on the configured `$install_options`, which means that it will need updating if the `$install_options` change. For example, if another `{ 'option' => 'value' }` gets added, the Tuple structure must be updated too. I wish this wasn't as rigorous but that's how it is.

On turning it into a custom datatype, please provide the code for that as I don't know how that should be done.

Reply:
okay, tuple doesn't need to be more complex. You can find an example for custom datatypes in our zabbix repo.
Types are defined in a `/types/` subdirectory:
https://github.com/voxpupuli/puppet-zabbix/blob/master/types/databases.pp
and used like this:
https://github.com/voxpupuli/puppet-zabbix/blob/master/manifests/server.pp#L296

That makes them reuseable but also shorter to write. so the enterprise.pp would be easier to read.

Reply:
OK, adding types in next commit